### ISSUE TITLE
Restarting an app is broken after updating to docker for mac to Version 17.12.0-ce-mac47 (21805)

### ISSUE THREAD

**Issue by Toshbrown** [me-box/databox#230]
After updating docker for mac to Version 17.12.0-ce-mac47 (21805)

Restarting an app is broken. Looks like a change in the networking behaviour of docker. 

On restarting the databox app it is assigned a new IP address and the core network then blocks access to the store.

d logs f82731a92307  -f (core-store)
```
[removed some data]
Netif: connect eth3
2018-01-18 11:40:27 +00:00: INF [junction] register intf eth3 10.0.3.3 10.0.3.0/24
2018-01-18 11:40:31 +00:00: INF [policy] Policy.connect skip privileged hostname app-os-monitor|arbiter
2018-01-18 11:40:31 +00:00: INF [dns] try to resolve app-os-monitor...
2018-01-18 11:40:31 +00:00: INF [dns] try to resolve export-service...
2018-01-18 11:40:31 +00:00: INF [dns] try to resolve app-os-monitor...
2018-01-18 11:40:31 +00:00: INF [dns] try to resolve driver-os-monitor-core-store...
2018-01-18 11:40:31 +00:00: INF [dns] resolved: export-service 10.0.0.5
2018-01-18 11:40:31 +00:00: INF [dns] resolved: driver-os-monitor-core-store 10.0.2.4
2018-01-18 11:40:40 +00:00: INF [dns] Dns_service: banned 10.0.3.4 to resolve driver-os-monitor-core-store
2018-01-18 11:40:40 +00:00: INF [policy] Policy.connect_for_privileged 10.0.3.4 <> arbiter
2018-01-18 11:40:40 +00:00: INF [dns] try to resolve arbiter...
2018-01-18 11:40:40 +00:00: INF [dns] resolved: arbiter 10.0.0.4
2018-01-18 11:40:40 +00:00: INF [interfaces] acquire fake ip 10.0.3.254 from eth3 10.0.3.0/24
2018-01-18 11:40:40 +00:00: INF [interfaces] acquire fake ip 10.0.0.250 from eth0 10.0.0.0/24
2018-01-18 11:40:40 +00:00: INF [policy] allow 10.0.3.4 to resolve arbiter (as 10.0.3.254)
2018-01-18 11:40:40 +00:00: INF [policy] add transport 10.0.3.4 -> 10.0.3.254
2018-01-18 11:40:40 +00:00: INF [policy] add transport 10.0.0.4 -> 10.0.0.250
2018-01-18 11:40:40 +00:00: INF [NAT] new NAT rule: (10.0.3.4 -> 10.0.3.254) => (10.0.0.250 -> 10.0.0.4)
2018-01-18 11:40:40 +00:00: INF [NAT] new NAT rule: (10.0.0.4 -> 10.0.0.250) => (10.0.3.254 -> 10.0.3.4)
2018-01-18 11:40:40 +00:00: INF [dns] Dns_service: banned 10.0.3.4 to resolve driver-os-monitor-core-store
2018-01-18 11:40:40 +00:00: INF [dns] Dns_service: banned 10.0.3.4 to resolve driver-os-monitor-core-store
2018-01-18 11:40:40 +00:00: INF [dns] Dns_service: banned 10.0.3.4 to resolve driver-os-monitor-core-store
2018-01-18 11:40:41 +00:00: INF [dns] resolved: app-os-monitor 10.0.3.4
2018-01-18 11:40:41 +00:00: INF [dns] resolved: app-os-monitor 10.0.3.4
2018-01-18 11:40:41 +00:00: INF [interfaces] acquire fake ip 10.0.3.252 from eth3 10.0.3.0/24
2018-01-18 11:40:41 +00:00: INF [interfaces] acquire fake ip 10.0.3.253 from eth3 10.0.3.0/24
2018-01-18 11:40:41 +00:00: INF [interfaces] acquire fake ip 10.0.2.251 from eth2 10.0.2.0/24
2018-01-18 11:40:41 +00:00: INF [policy] allow 10.0.3.4 to resolve driver-os-monitor-core-store (as 10.0.3.252)
2018-01-18 11:40:41 +00:00: INF [policy] allow 10.0.2.4 to resolve app-os-monitor (as 10.0.2.251)
2018-01-18 11:40:41 +00:00: INF [policy] add transport 10.0.3.4 -> 10.0.3.252
2018-01-18 11:40:41 +00:00: INF [policy] add transport 10.0.2.4 -> 10.0.2.251
2018-01-18 11:40:41 +00:00: INF [NAT] new NAT rule: (10.0.3.4 -> 10.0.3.252) => (10.0.2.251 -> 10.0.2.4)
2018-01-18 11:40:41 +00:00: INF [NAT] new NAT rule: (10.0.2.4 -> 10.0.2.251) => (10.0.3.252 -> 10.0.3.4)
2018-01-18 11:40:41 +00:00: INF [policy] Policy.connect app-os-monitor <> driver-os-monitor-core-store
2018-01-18 11:40:41 +00:00: INF [interfaces] acquire fake ip 10.0.0.249 from eth0 10.0.0.0/24
2018-01-18 11:40:41 +00:00: INF [policy] allow 10.0.3.4 to resolve export-service (as 10.0.3.253)
2018-01-18 11:40:41 +00:00: INF [policy] allow 10.0.0.5 to resolve app-os-monitor (as 10.0.0.249)
2018-01-18 11:40:41 +00:00: INF [policy] add transport 10.0.3.4 -> 10.0.3.253
2018-01-18 11:40:41 +00:00: INF [policy] add transport 10.0.0.5 -> 10.0.0.249
2018-01-18 11:40:41 +00:00: INF [NAT] new NAT rule: (10.0.3.4 -> 10.0.3.253) => (10.0.0.249 -> 10.0.0.5)
2018-01-18 11:40:41 +00:00: INF [NAT] new NAT rule: (10.0.0.5 -> 10.0.0.249) => (10.0.3.253 -> 10.0.3.4)
2018-01-18 11:40:41 +00:00: INF [policy] Policy.connect app-os-monitor <> export-service

[Restart the OS minitor app]

2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:20 +00:00: INF [policy] Policy.connect_for_privileged 10.0.3.5 <> arbiter
2018-01-18 11:41:20 +00:00: INF [dns] try to resolve arbiter...
2018-01-18 11:41:20 +00:00: INF [dns] resolved: arbiter 10.0.0.4
2018-01-18 11:41:20 +00:00: INF [interfaces] acquire fake ip 10.0.3.250 from eth3 10.0.3.0/24
2018-01-18 11:41:20 +00:00: INF [interfaces] acquire fake ip 10.0.0.247 from eth0 10.0.0.0/24
2018-01-18 11:41:20 +00:00: INF [policy] allow 10.0.3.5 to resolve arbiter (as 10.0.3.250)
2018-01-18 11:41:20 +00:00: INF [policy] add transport 10.0.3.5 -> 10.0.3.250
2018-01-18 11:41:20 +00:00: INF [policy] add transport 10.0.0.4 -> 10.0.0.247
2018-01-18 11:41:20 +00:00: INF [NAT] new NAT rule: (10.0.3.5 -> 10.0.3.250) => (10.0.0.247 -> 10.0.0.4)
2018-01-18 11:41:20 +00:00: INF [NAT] new NAT rule: (10.0.0.4 -> 10.0.0.247) => (10.0.3.250 -> 10.0.3.5)
2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:20 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:21 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:21 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:21 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
2018-01-18 11:41:21 +00:00: INF [dns] Dns_service: banned 10.0.3.5 to resolve driver-os-monitor-core-store
```

**Comment by sevenEng** [comment]
Got it, will work on a patch to deal with this.

**Comment by sevenEng** [comment]
seems not only docker for mac, after updating on my xubuntu machine, the service's IP will also change after a restart

**Comment by Toshbrown** [comment]
Fixed in 0.3.2
